### ISSUE TITLE
multi-touch views

### ISSUE THREAD

**Issue by rgangopadhya** [kmagiera/react-native-gesture-handler#47]
By this I mean something like:

```
<View
    onTouchStart={() => { console.log('pressing left button'); }}
    onTouchEnd={() => { console.log('releasing left button'); }}
>
    {...}
</View>
<View
    onTouchStart={() => { console.log('pressing right button'); }}
    onTouchEnd={() => { console.log('releasing right button'); }}
>
    {...}
</View>
```

The idea is user has to hold both the left and right button for something to be triggered.

This works fine on iOS but not on Android: related to https://github.com/facebook/react-native/issues/10068.

So, was hoping I could use this library to work around that issue. Doesn't seem to work however -- replacing the above with 

```
<BaseButton
  onActiveStateChange={(isActive) => {
    if (isActive) {
      console.log('pressing left button');
    } else {
      console.log('releasing left button');
    }
  }}
>
  {...}
</BaseButton>
<BaseButton
  onActiveStateChange={(isActive) => {
    if (isActive) {
      console.log('pressing right button');
    } else {
      console.log('releasing right button');
    }
  }}
>
  {...}
</BaseButton>
```

But this results in the same issue as using `View`: once one button is held, pressing the other triggers the original button's handler. 

Is there any way to use `waitFor` here?

**Comment by yossarin** [comment]
Hello,

I am curious what is the obstacle with implementing multi-touch handlers in android?
(Obviously I need this feature)

**Comment by yossarin** [comment]
I am asking because I want to help with this issue, maybe you have some pointers for me? :) @kmagiera

**Comment by ryan4664** [comment]
I'm still having this issue on Android but have no issues on iOS.  I'm on the latest version.  Any suggestions?

**Comment by jacksonlevine** [comment]
> Hey @yossarin @rgangopadhya
> 
> Just letting you know we've merged PR that adds support for that: [4036fba](https://github.com/software-mansion/react-native-gesture-handler/commit/4036fba2fbc7e05d1fea340676a2975594936e8c)
> 
> The demo is here: https://twitter.com/kzzzf/status/1015907930649055233
> 
> After quite a long time I am happy to finally close this issue. If you have any questions or want to discuss this further feel free to reopen it or create a new issue.

Hey, is there any documentation on how this can be done? I see you achieved it but don't see how to do it

**Comment by kmagiera** [comment]
hey @jacksonlevine since it's been a while since this issue was opened and support for multi-touch got added, it'd be good if you could be more precise to describe your usecase and also consider opening a new issue to track it if it is about something different than this issue.

As for the effect in the demo, there is nothing specific that needs to be added. If you have gesture handlers that are allowed to run in parallel you should be able to use additional fingers to control them. This is in line with the behavior on iOS now.

**Comment by sakibahmedshanto** [comment]
I was facing the same issue and after a lot of trial and error, 
  I found that React Native Gesture Handler solves this perfectly!
  
  Steps:
  1. Install: npm install react-native-gesture-handler
  2. Import: import { Gesture, GestureDetector } from 'react-native-gesture-handler';
  3. Create gesture:
  
  const pressGesture = Gesture.LongPress()
    .minDuration(0) // Immediate response
   .onStart(() => {
      'worklet';
      runOnJS(handlePressStart)();
    })
   .onEnd(() => {
     'worklet';
     runOnJS(handlePressEnd)();
    });
  
  4. Wrap your component: with =>    GestureDetector gesture={pressGesture}
 
  
  ✅ Now multiple components can be pressed simultaneously!
  Perfect for game controllers, multi-touch interfaces, etc.

give this as reference to ai and take help from the ai

**Comment by noubar** [comment]
@sakibahmedshanto 
Thank you alot! very very helpful.

I have only one small note to add:

For example i have two elements (element1, element2), both need to be touchable simoultaneously. Then both elements should be part of the same GestureHandlerRootView.

```
<GestureHandlerRootView
  <GestureDetector gesture={pressgesture1}>
    <Element1>
    </Element1>
  </GestureDetector>
  <GestureDetector gesture={pressgesture2}>
    <Element2>
    </Element2>
  </GestureDetector>
</GestureHandlerRootView>
```
